src/controller/AutoCommissioner.cpp

```diff
@@ -258,10 +260,22 @@ CHIP_ERROR AutoCommissioner::StartCommissioning(DeviceCommissioner * commissione
 
 Optional<System::Clock::Timeout> AutoCommissioner::GetCommandTimeout(CommissioningStage stage)
 {
-    // Per spec, all commands that are sent with the arm failsafe held need at least a 30s timeout. Using 30s everywhere for
-    // simplicity. 30s appears to be sufficient for long-running network commands (enable wifi and enable thread), but we may wish
-    // to increase the timeout for those commissioning stages at a later time.
-    return MakeOptional(System::Clock::Timeout(System::Clock::Seconds16(30)));
+    // Per spec, all commands that are sent with the arm failsafe held need at least a 30s timeout.
+    // Network clusters can indicate the time required to connect, so if we are connecting, use that time as long as it is > 30s.
+    uint8_t seconds = 30;
```

Comment:
Can we get this type with cluster-object.h with this?

```suggestion
    NetworkCommissioning::Attributes::ConnectMaxTimeSeconds::TypeInfo::Type seconds = 30;
```